marklogic-client-api/src/main/java/com/marklogic/client/datamovement/ZipSplitter.java

```diff
@@ -0,0 +1,363 @@
+/*
+ * Copyright 2019 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.marklogic.client.datamovement;
+
+import com.marklogic.client.document.DocumentWriteOperation;
+import com.marklogic.client.impl.DocumentWriteOperationImpl;
+import com.marklogic.client.io.BytesHandle;
+import com.marklogic.client.io.Format;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+/**
+ * The ZipSplitter class is used to split compressed files.
+ * The zip file could include XML, JSON, TXT or BINARY files.
+ */
+public class ZipSplitter implements Splitter<BytesHandle> {
+    private Map<String, Format> extensionFormats;
+    private Predicate<ZipEntry> entryFilter;
+    private Function<String, String> uriTransformer;
+    private int count = 0;
+
+    /**
+     * Returns the extensionFormats set to splitter. The extensionFormat is pre-defined in splitter.
+     * It includes "json", "txt" and "xml" extensions. If the file has no extension, it is treated as binary file.
```

Comment:
Could the JavaDoc also let the developer know that she can add mappings from other extensions of documents in the zipfile to one of the four MarkLogic formats?

Reply:
Add below description.
"You can also add mappings from other extensions in the zipfile to one of the four MarkLogic formats."